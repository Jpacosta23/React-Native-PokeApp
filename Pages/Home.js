import React, { useState } from "react";
import {
  Button,
  View,
  Text,
  StyleSheet,
  Image,
  TouchableOpacity,
} from "react-native";

function HomeScreen({ navigation }) {
  const [pokemon, setPokemon] = useState(null);
  const [pokePicture, setPokePicture] = useState("");

  const handlePress = async () => {
    try {
      let random = Math.round(Math.random() * 151);
      const response = await fetch(
        `https://pokeapi.co/api/v2/pokemon/${random}`
      );
      const data = await response.json();
      setPokemon(data);
      setPokePicture(data.sprites.front_default);
    } catch (error) {
      console.log(error);
    }
  };
  return (
    <View style={styles.main}>
      <View style={styles.container}>
        <Image
          source={require("../assets/Imgs/logo.png")}
          style={{ width: 300, height: 110 }}
        />
        <Text style={styles.title}>Catch them now</Text>
        <View style={styles.constainerCol}>
          <TouchableOpacity style={styles.button} onPress={() => handlePress()}>
            <Text style={styles.buttonText}>Find Pokemon</Text>
          </TouchableOpacity>
          {pokemon ? (
            <TouchableOpacity
              style={styles.button}
              onPress={() => {
                alert(pokemon.name);
                setPokemon(null);
              }}
            >
              <Text style={styles.buttonText}>Throw pokeball</Text>
            </TouchableOpacity>
          ) : null}
        </View>
      </View>

      <View style={styles.containerRow}>
        {!pokemon ? (
          <Image
            source={require("../assets/Imgs/Pokeball_icon-icons.com_67533.png")}
            style={{ width: 120, height: 120 }}
          />
        ) : (
          <Image
            source={
              pokePicture
                ? { uri: pokePicture }
                : require("../assets/Imgs/Pokeball_icon-icons.com_67533.png")
            }
            style={{ width: 140, height: 140 }}
          />
        )}
      </View>
    </View>
  );
}

export default HomeScreen;

const styles = StyleSheet.create({
  main: {
    flex: 1,
    backgroundColor: "red",
    alignItems: "center",
    justifyContent: "space-around",
  },
  container: {
    flex: 4,
    alignItems: "center",
    justifyContent: "space-around",
  },
  containerRow: {
    flex: 1,
    alignItems: "center",
    flexDirection: "row",
    justifyContent: "center",
    width: 300,
  },
  constainerCol: {
    height: 200,
    justifyContent: "space-around",
  },
  title: {
    fontSize: 30,
    fontFamily: "Minecraft-regular",
  },
  button: {
    alignItems: "center",
    justifyContent: "center",
    paddingVertical: 12,
    paddingHorizontal: 32,
    borderRadius: 4,
    elevation: 3,
    backgroundColor: "white",
  },
  buttonText: {
    fontSize: 20,
    fontFamily: "Minecraft-regular",
    color: "red",
  },
});
