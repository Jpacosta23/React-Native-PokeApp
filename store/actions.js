import { getPokemon } from "../services";

/**
 * Get a single Job
 * @param {Number} id
 * @param {Function} dispatch
 */
const getPokemonDetail = async (id, dispatch) => {
  dispatch({ type: "IS_LOADING", payload: true });
  try {
    const pokemon = await getPokemon(id);
    dispatch({ type: "GET_POKEMON", payload: pokemon });
  } catch (error) {
    dispatch({ type: "SHOW_ERROR", payload: error });
  } finally {
    dispatch({ type: "IS_LOADING", payload: false });
  }
};

const deletePokemonDetail = (dispatch) => {
  dispatch({ type: "IS_LOADING", payload: true });
  try {
    dispatch({ type: "RESET_POKEMON", payload: null });
  } catch (error) {
    dispatch({ type: "SHOW_ERROR", payload: error });
  } finally {
    dispatch({ type: "IS_LOADING", payload: false });
  }
};

const capturePokemon = (pokemon, dispatch) => {
  try {
    dispatch({ type: "CAPTURE_POKEMON", payload: pokemon });
  } catch (error) {
    dispatch({ type: "SHOW_ERROR", payload: error });
  }
};

const releasePokemon = (pokemon, dispatch) => {
  try {
    dispatch({ type: "RELEASE_POKEMON", payload: pokemon });
  } catch (error) {
    dispatch({ type: "SHOW_ERROR", payload: error });
  }
};

export {
  getPokemonDetail,
  deletePokemonDetail,
  capturePokemon,
  releasePokemon,
};
