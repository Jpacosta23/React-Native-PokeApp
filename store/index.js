import React, { createContext, useReducer, useContext } from "react";

const initialState = {
  pokemon: null,
  pokedex: [],
  pokebag: [],
  isLoading: true,
  showError: false,
};

const AppStateContext = createContext();
const AppDispatchContext = createContext();

const AppReducer = (state, action) => {
  switch (action.type) {
    case "GET_POKEMON": {
      return {
        ...state,
        pokemon: action.payload,
      };
    }
    case "RESET_POKEMON": {
      return {
        ...state,
        pokemon: action.payload,
      };
    }
    case "SHOW_ERROR": {
      return {
        ...state,
        showError: true,
        message: action.payload,
      };
    }
    case "IS_LOADING": {
      return {
        ...state,
        isLoading: action.payload,
      };
    }
    default: {
      return state;
    }
  }
};

const AppProvider = ({ children }) => {
  const [state, dispatch] = useReducer(AppReducer, initialState);
  return (
    <AppStateContext.Provider value={state}>
      <AppDispatchContext.Provider value={dispatch}>
        {children}
      </AppDispatchContext.Provider>
    </AppStateContext.Provider>
  );
};

const useAppState = () => {
  const context = useContext(AppStateContext);
  if (context === undefined) {
    throw new Error("useAppState must be used withim a AppProvider");
  }

  return context;
};

const useAppDispatch = () => {
  const context = useContext(AppDispatchContext);
  if (context === undefined) {
    throw new Error("useAppDispatch must be used within a AppProvider");
  }

  return context;
};

export { AppProvider, useAppState, useAppDispatch };
