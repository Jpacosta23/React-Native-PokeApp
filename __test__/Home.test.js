import React from "react";
import "react-native";
import Home from "../Pages/Home";
import renderer from "react-test-renderer";
import { AppProvider } from "../store";

test("Home snapShot", async () => {
  beforeEach(() => {
    global.fetch = jest.fn(() =>
      Promise.resolve({
        json: () => Promise.resolve(),
      })
    );
  });
  const snap = await renderer
    .create(
      <AppProvider>
        <Home />
      </AppProvider>
    )
    .toJSON();

  expect(snap).toMatchSnapshot();
});
