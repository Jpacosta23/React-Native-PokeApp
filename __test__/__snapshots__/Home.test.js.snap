// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapShot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "red",
      "flex": 1,
      "justifyContent": "space-around",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 4,
        "justifyContent": "space-around",
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 110,
          "width": 300,
        }
      }
    />
    <Text
      style={
        Object {
          "fontFamily": "Minecraft-regular",
          "fontSize": 30,
        }
      }
    >
      Catch them now
    </Text>
    <View
      style={
        Object {
          "height": 200,
          "justifyContent": "space-around",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 4,
            "elevation": 3,
            "justifyContent": "center",
            "opacity": 1,
            "paddingHorizontal": 32,
            "paddingVertical": 12,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "red",
              "fontFamily": "Minecraft-regular",
              "fontSize": 20,
            }
          }
        >
          Find Pokemon
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "width": 300,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 120,
          "width": 120,
        }
      }
    />
  </View>
</View>
`;
